Compfiles/Imo2003P1.lean

```diff
@@ -23,12 +23,115 @@ are pairwise disjoint.
 
 namespace Imo2003P1
 
-abbrev S : Finset ℕ := Finset.Icc 1 (10^6)
+abbrev S := Finset.Icc 1 1000000
 
-problem imo2003_p1 (A : Finset ℕ) (hA : A ⊆ S) :
-    ∃ t : Fin 100 → S,
-      ∀ i j, i ≠ j → Disjoint {x + t i | x ∈ A} {x + t j | x ∈ A} := by
-  sorry
+/- Proof by induction on the number of tᵢ's -/
+theorem induction_lemma (A : Finset ℕ) (AS : A ⊆ S) (Acard : A.card = 101) {k : ℕ} (hk : k ≤ 100) :
```

Comment:
Please put helper lemmas like this in a `snip begin` ... `snip end` block. https://github.com/dwrensha/compfiles?tab=readme-ov-file#extracting-problems